backend/e4h-services/ingestion-service/app/api/endpoints/file_ingestion.py

```
@@ -944,4 +967,379 @@ async def upload_facility_selection_excel_sheet(
+    return FileResponse(output_path, filename=os.path.basename(output_path), media_type="application/vnd.openxmlformats-officedocument.spreadsheetml.sheet")
+
+
+@router.post("/check_duplicates")
+async def check_duplicate_tickets(
+        legacy_ticket_file: UploadFile = File(...),
+        legacy_ticket_sheet_name: str = Form(default="Duplication Template"),
+):
+    input_temp_file = None
+    try:
+        # Save uploaded file temporarily
+        input_temp_file = tempfile.NamedTemporaryFile(delete=False, suffix=".xlsx")
```

Ruff (0.11.9)

Use a context manager for opening files

(SIM115)